crates/cast/bin/cmd/wallet/mod.rs

```diff
@@ -308,13 +317,33 @@ impl WalletSubcommands {
                 let wallets =
                     wallets.into_iter().map(|b| b.build()).collect::<Result<Vec<_>, _>>()?;
 
-                println!("{}", "Successfully generated a new mnemonic.".green());
-                println!("Phrase:\n{phrase}");
-                println!("\nAccounts:");
+                if !json {
+                    println!("{}", "Successfully generated a new mnemonic.".green());
+                    println!("Phrase:\n{phrase}");
+                    println!("\nAccounts:");
+                }
+
+                let mut accounts = json!([]);
                 for (i, wallet) in wallets.iter().enumerate() {
-                    println!("- Account {i}:");
-                    println!("Address:     {}", wallet.address());
-                    println!("Private key: 0x{}\n", hex::encode(wallet.credential().to_bytes()));
+                    let private_key = hex::encode(wallet.credential().to_bytes());
+                    if json {
+                        accounts.as_array_mut().unwrap().push(json!({
+                            "address": wallet.address(),
```

Comment:
maybe do `"address": format!("{}", wallet.address()),` to preserve check-summed format

Reply:
Addressed in 6179e125b17c7ab748a0c53806c71ac1b2e99222